packages/block-library/src/button/editor.scss

```diff
@@ -63,7 +63,6 @@
 	.is-selected & {
 		height: auto;
 		overflow: visible;
-		margin-top: $grid-unit-20;
```

Comment:
Do you know what this margin was for?

Reply:
It was added [here](https://github.com/WordPress/gutenberg/pull/10128/files), looks like it was applied to the `LinkControl` in the block toolbar. It's been refactored since then and I'm fairly confident this isn't getting used at all anymore, actually 🤔

I have tested adding/removing a button's link and the controls look good to me.